java/core/src/main/java/com/microsoft/bond/Bonded.java

```diff
@@ -135,6 +135,19 @@
      */
     public abstract <U extends BondSerializable> Bonded<U> convert(StructBondType<U> toBondType);
 
+    public <U extends BondSerializable> Bonded<U> cast(StructBondType<U> castBondType) {
+        if (!(getBondType().isSubtypeOf(castBondType))) {
+            final String err = String.format("Bonded<%s> cannot be cast to Bonded<%s>",
+                getBondType().getQualifiedName(), castBondType.getQualifiedName()
+            );
+            throw new ClassCastException(err);
```

Comment:
`cast` throws, but `convert` returns null? Is the plan to change `convert` to throw too? (It has this behavior in the two object-backed implementations at the moment.) Should that happen now?